privacyidea/lib/eventhandler/responsemangler.py

```diff
@@ -0,0 +1,164 @@
+import json
+import logging
+
+
+log = logging.getLogger(__name__)
+
+
+class ACTION_TYPE(object):
+    """
+    Allowed actions
+    """
+    ADD = "add"
+    DELETE = "delete"
+
+
+class ResponseManglerHandler(BaseEventHandler):
+    """
+    An Eventhandler needs to return a list of actions, which it can handle.
+
+    It also returns a list of allowed action and conditions
+
+    It returns an identifier, which can be used in the eventhandlig definitions
+    """
+
+    identifier = "ResponseMangler"
+    description = "This event handler can mangle the JSON response."
+
+    @property
+    def allowed_positions(cls):
+        """
+        This returns the allowed positions of the event handler definition.
+        The ResponseMangler can only be located at the "post" position
+
+        :return: list of allowed positions
+        """
+        return ["post"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        actions = {ACTION_TYPE.DELETE:
+                       {"JSON pointer":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The JSON pointer (key) that should be deleted. Please "
+                                              "specify in the format '/detail/message'.")}
+                        },
+                   ACTION_TYPE.ADD:
+                       {"JSON pointer":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The JSON pointer (key) that should be added. "
+                                              "Please specify in the format '/detail/message'.")
+                             },
+                        "type":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The type of the value."),
+                             "value": ["string", "integer", "bool"]
+                            },
+                        "value":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The value of the JSON key that should be added.")
+                            }
+                        }
+                   }
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action:
+        :param options: Contains the flask parameters g, request, response
+            and the handler_def configuration
+        :type options: dict
+        :return:
+        """
+        ret = True
+        response = options.get("response")
+        content = self._get_response_content(response)
```

Comment:
We should check if we got a dictionary here before manipulating it.
Actually i broke this in 3fdac34 since i try to enforce parsing the `data` attribute as JSON.

Reply:
TO my understanding this is not relevant anymore, since we catched it further upwards, if it is not json.

(see)

The ``self._get_response_content(response)`` actually allows parsing either real json response or text/html as json. 
So I decided to catch the exception in case the response is not jsonable.
Done in 0e16f5f5